samples/besu/simpleasset/app/AssetExchangeERC20.js

```diff
@@ -0,0 +1,308 @@
+async function getContractInstance(provider, pathToJson){
+
+	const jsonFile = path.resolve(__dirname, pathToJson)
+	var jsonFileContents = fs.readFileSync(jsonFile)
+	var contractName = contract(JSON.parse(jsonFileContents))
+	contractName.setProvider(provider)
+	var instance = await contractName.deployed()
+
+	return instance
+}
+
+
+// Initialization of the parameters
+async function init(provider1, provider2, contractOwner1, contractOwner2, Alice1, Bob2, tokenSupply, senderInitialBalance) {
+	
+    	htlc1 = await getContractInstance(provider1,'../build/contracts/InteroperationBaseClassERC20.json') 
+    	htlc2 = await getContractInstance(provider2,'../build/contracts/InteroperationBaseClassERC20.json') 
+
+	AliceERC20 = await getContractInstance(provider1, '../build/contracts/AliceERC20.json')
+	
+	BobERC20 = await getContractInstance(provider2, '../build/contracts/BobERC20.json')
+
+// ALREADY RUN ONCE: So commenting it out for now 
+    await AliceERC20.transfer(Alice1, senderInitialBalance, {from: contractOwner1}) // so Alice has some tokens to trade in Network 1
+    await BobERC20.transfer(Bob2, senderInitialBalance, {from: contractOwner2}) // so Bob has some tokens to trade in Network 2
+//*/
+
+// Asserts have to be rewritten
+	/*
+    await assertTokenBal(
+      AliceERC20,
+      Alice,
+      senderInitialBalance,
+      'balance not transferred to Alice in before()'
+    )
+    await assertTokenBal(
+      AliceERC20,
+      Bob,
+      0,
+      'Bob should not have any AliceERC20 tokens in before()'
+    )
+    await assertTokenBal(
+      BobERC20,
+      Bob,
+      senderInitialBalance,
+      'balance not transferred to Bob in before()'
+    )
+    await assertTokenBal(
+      BobERC20,
+      Alice,
+      0,
+      'Alice should not have any BobERC20 tokens in before()'
+    )
+*/
+}
+
+
+// Withdrawal function of a fungible asset/token of type tokenContractId 
+// and amount tokenAmount locked using contractId for a receiver. 
+// hashSecret is the secret to unlock the asset corresponding to the commitment 
+// used when locking it
+async function withdraw(htlc, contractId, tokenContractId, tokenAmount, receiver, hashSecret) {
+    await htlc.claimFungibleAsset(contractId, hashSecret, {
+      from: receiver,
+    })
+
+/*
+    // Check tokens now owned by Alice
+    await assertTokenBal(
+      tokenContractId,
+      receiver,
+      tokenAmount,
+      `${receiver} doesn't not own ${tokenAmount} tokens`
+    )
+*/
+
+    const contractArr = await htlc.getContract.call(contractId)
+    const contract = htlcERC20ArrayToObj(contractArr)
+    assert.isTrue(contract.withdrawn) // withdrawn set
+    assert.isFalse(contract.refunded) // refunded still false
+    assert.equal(contract.preimage, hashSecret)
+
+	return contract.withdrawn // the withdrawn flag is set iff the withdraw function is successful
+}
+
+
+const assertTokenBal = async (token, addr, tokenAmount, msg) => {
```

Comment:
We have a mix of anonymous (e.g., line 126) and named (e.g., line 135) functions. Any particular reason for that? I don't have a strong preference for either, though debugging may be easier with named functions.